### PR TITLE
HLM-5154 adding boundary selection and its reflection on map

### PR DESCRIPTION
**Pull request by siddhant-nawale-egov**

### CODE REVIEW

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/pages/employee/MicroplanPreview.js [hunk 1]
```diff
@@ -90,7 +90,8 @@ const MicroplanPreview = ({
   // Set TourSteps
   useEffect(() => {
     const tourData = tourSteps(t)?.[page] || {};
-    if (!state?.tourStateData?.name || state.tourStateData.name === page) return;
+    console.log(tourData, state)
+    if (state?.tourStateData?.name === page) return;
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/pages/employee/MicroplanPreview.js]: 1. The error handling has been done. and we are using mutate which has a error handling of its own.
2. fetchMicroplanPreviewData involves reading all the data that was uploaded by the user

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/pages/employee/Upload.js [hunk 2]
```diff
@@ -82,6 +82,7 @@ const Upload = ({
       BoundaryTypeHierarchySearchCriteria: {
         tenantId: Digit.ULBService.getStateId(),
         hierarchyType: campaignData?.hierarchyType,
+        // hierarchyType:  "Microplan",
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/pages/employee/Upload.js:85]: We are seting toast messages for specific errors.

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/pages/employee/Hypothesis.js [hunk 3]
```diff
@@ -43,15 +43,27 @@ const Hypothesis = ({ campaignType = "SMC", microplanData, setMicroplanData, che
...
+    if (!hypothesisAssumptions) return;
+    let temp = hypothesisAssumptions.find((item) => item.campaignType === campaignType);
+    if (!(temp && temp.assumptions)) return;
+    let hypothesisAssumptionsList = temp.assumptions
+    setExampleOption(temp.length ? temp[0] : "");
+
+    let newAssumptions = setAutofillHypothesisData(hypothesisAssumptionsList, microplanData?.hypothesis ? microplanData?.hypothesis : assumptions, setAssumptions);
+    let newHypothesislist = filterHypothesisList(newAssumptions.length !==0 ? newAssumptions : microplanData.hypothesis, hypothesisAssumptionsList);
+    setHypothesisAssumptionsList(newHypothesislist);
+
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/pages/employee/Hypothesis.js]: It only runs on first render

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/pages/employee/RuleEngine.js [hunk 4]
```diff
@@ -51,7 +51,8 @@ const RuleEngine = ({ campaignType = "SMC", microplanData, setMicroplanData, che
       const hypothesisAssumptions = microplanData?.hypothesis?.filter((item) => item.key !== "").map((item) => item.key) || [];
       if (hypothesisAssumptions.length !== 0) {
         setHypothesisAssumptionsList(hypothesisAssumptions);
-        setRules(microplanData.ruleEngine);
+        setRuleEngineDataFromSsn(microplanData.ruleEngine, hypothesisAssumptions, setRules);
+        // setRules(microplanData.ruleEngine);
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/pages/employee/RuleEngine.js]: I want it to run only on first render.

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/pages/employee/Mapping.js [hunk 5]
```diff
@@ -613,65 +652,69 @@ const extractGeoData = (
...
-          if (entityValue?.children) tempStorage = [...tempStorage, ...fetchFeatures(entityValue?.children, parameter, outputList)];
-        }
+      if (parameter.includes(entityKey) && entityValue && entityValue.data && entityValue.data.feature) {
+        let feature = entityValue.data.feature;
+        feature.properties["name"] = entityKey;
+        feature.properties["style"] = style;
+        if (entityValue?.children) tempStorage = [...tempStorage, feature, ...fetchFeatures(entityValue?.children, parameter, outputList, style)];
+        else tempStorage = [...tempStorage, feature];
+      } else {
+        if (entityValue?.children) tempStorage = [...tempStorage, ...fetchFeatures(entityValue?.children, parameter, outputList, style)];
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/pages/employee/Mapping.js]: It need to be this way as it is working on a tree and we wanted the function to be capable of fetching everything and custom entries depending on where it is used.

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/index.js [hunk 6]
```diff
@@ -157,16 +156,18 @@ const inputScrollPrevention = (e) => {
...
+      if (item?.error) return;
+      const data = { filestoreId: item.filestoreId, inputFileType: item.fileType, templateIdentifier: item.section };
+      files.push(data);
+    });
+    Object.values(data?.upload).forEach((item) => {
+      if (item?.error) return;
+      resourceMapping.push(item?.resourceMapping);
+    });
+    resourceMapping = resourceMapping.flatMap((inner) => inner);
+  }
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/index.js:170]: This removal doesnt affect the other parts. its just for api body preparation.

**Review by nipunarora-eGov — APPROVED** [review]

**Pull request merged**